src/openrct2/platform/posix.c

```diff
@@ -391,6 +391,10 @@ sint32 platform_enumerate_files_begin(const utf8 *pattern)
 				safe_strcpy(paths[idx], dir_name, path_len);
 				safe_strcat_path(paths[idx], d->d_name, path_len);
 				log_verbose("paths[%d] = %s", idx, paths[idx]);
+
+				#ifdef __MACOSX__
+					paths[idx] = macos_str_decomp_to_precomp(paths[idx]);
```

Comment:
Hmmm... I suppose that'd depend on whether the precomposed string ends up being longer than the decomposed one. I can't think of any cases where that'd be the case, but either way, probably best to use `safe_strcpy()` just in case.